packages/flutter/lib/src/painting/decoration.dart

```diff
@@ -44,27 +44,72 @@ abstract class Decoration extends Diagnosticable {
   /// of the decoration. For example, if the decoration draws a frame
   /// around its edge, the padding would return the distance by which
   /// to inset the children so as to not overlap the frame.
-  EdgeInsets get padding => null;
+  ///
+  /// This only works for decorations that have absolute sizes. If the padding
+  /// needed would change based on the size at which the decoration is drawn,
+  /// then this will return incorrect padding values.
+  ///
+  /// For example, when a [BoxDecoration] has [BoxShape.circle], the padding
+  /// does not take into account that the circle is drawn in the center of the
+  /// box regardless of the ratio of the box; it does not provide the extra
+  /// padding that is implied by changing the ratio.
+  EdgeInsets get padding => EdgeInsets.zero;
 
   /// Whether this decoration is complex enough to benefit from caching its painting.
   bool get isComplex => false;
 
-  /// Linearly interpolates from [a] to [this].
-  Decoration lerpFrom(Decoration a, double t) => this;
+  /// Linearly interpolates from `a` to [this].
+  ///
+  /// When implementing this method in subclasses, return null if this class
+  /// cannot interpolate from `a`. In that case, [lerp] will try `a`'s [lerpTo]
+  /// method instead.
+  ///
+  /// Supporting interpolating from null is recommended as the [Decoration.lerp]
+  /// method uses this as a fallback when two classes can't interpolate between
+  /// each other.
+  ///
+  /// Instead of calling this directly, use [Decoration.lerp].
+  @protected
+  Decoration lerpFrom(Decoration a, double t) => null;
```

Comment:
Can you document `t` in these?

Reply:
This affects a ton of code so filed https://github.com/flutter/flutter/issues/12377.